frontend/packages/console-plugin-sdk/src/typings/dev-catalog.ts

```diff
@@ -1,17 +1,115 @@
 import { K8sKind, K8sResourceKind } from '@console/internal/module/k8s';
-import { Extension } from './base';
+import { CodeRef, Extension } from './base';
 
 namespace ExtensionProperties {
   export interface DevCatalogModel {
     model: K8sKind;
     normalize: (data: K8sResourceKind[]) => K8sResourceKind[];
   }
+
+  export interface CatalogItemType {
+    /** Type for the catalog item. */
+    type: string;
+    /** Title fpr the catalog item. */
+    title: string;
+    /** Catalog ID for which this provider contributes to. Idea here is to be able to support multiple catalogs. */
+    catalog: string;
+    /** Description for the type specific catalog. */
+    catalogDescription?: string;
+    /** Description for the catalog item type. */
+    typeDescription?: string;
+    /** Custom filters specific to the catalog item  */
+    filters?: CodeRef<CatalogExtensionHook<CatalogFilter[]>>;
+    /** Custom groupings specific to the catalog item */
+    groupings?: CodeRef<CatalogExtensionHook<CatalogGrouping[]>>;
+  }
+
+  export interface CatalogItemProvider {
+    /** Type ID for the catalog item type. */
+    type: string;
+    /** Fetch items and normalize it for the catalog. Value is a react effect hook. */
+    provider: CodeRef<CatalogExtensionHook<CatalogItem[]>>;
+  }
 }
 
 export interface DevCatalogModel extends Extension<ExtensionProperties.DevCatalogModel> {
   type: 'DevCatalogModel';
 }
 
+export interface CatalogItemType extends Extension<ExtensionProperties.CatalogItemType> {
+  type: 'Catalog/ItemType';
+}
+
+export interface CatalogItemProvider extends Extension<ExtensionProperties.CatalogItemProvider> {
+  type: 'Catalog/ItemProvider';
+}
+
 export const isDevCatalogModel = (e: Extension): e is DevCatalogModel => {
   return e.type === 'DevCatalogModel';
 };
+
+export const isCatalogItemType = (e: Extension): e is CatalogItemType => {
+  return e.type === 'Catalog/ItemType';
+};
+
+export const isCatalogItemProvider = (e: Extension): e is CatalogItemProvider => {
+  return e.type === 'Catalog/ItemProvider';
+};
+
+export type CatalogExtensionHookResult<T> = [T, boolean, any];
+
+export type CatalogExtensionHookOptions = {
+  namespace: string;
+};
+
+export type CatalogExtensionHook<T> = (
+  options: CatalogExtensionHookOptions,
+) => CatalogExtensionHookResult<T>;
+
+export type CatalogItem = {
+  uid: string;
+  type: string;
+  name: string;
+  displayName?: string;
```

Comment:
For helm charts we need name of the chart to filter out different versions of the same chart. Earlier there was always obj.metadata.name to refer to for any such use cases but now since we don't have the obj I thought it would be good to have both name and displayName.

Reply:
Removed displayName and using attributes.name instead.